forem/forem
# Add tracking for the registration flow - "Clicked on Create Account"

## What type of PR is this? (check all applicable)


## Description

This PR tackles adding tracking for the registration flow.  The elements that need to be tracked are outlined in this [issue](https://github.com/forem/forem-internal-eng/issues/490). 

There's been discussion about the structure and keys needed of the data in [the auditing issue](https://github.com/forem/forem-internal-eng/issues/518#issuecomment-1246739027). 

Heres a List of events with some further information: 
| source               | trigger                  | referring_source  |
|----------------------|--------------------------|-------------------|
| top_navbar           | N/A                      | N/A               |
| left_sidebar         | N/A                      | N/A               |
| mobile_hamburger_nav | N/A                      | N/A               |
| feed_card            | N/A                      | N/A               |
| modal                | like/unicorn/readinglist | reactions_toolbar |
| modal                | like                     | comment           |
| modal                | reply                    | comment           |
| modal                | comment_subscription     | comments          |
| modal                | new_comment              | comments          |
| modal                | post_index_toolbar       | readinglist       |
| modal                | user                     | follow_button     |
| modal                | search                   | my_posts_filter   |
|                      |                          |                   |

## Related Tickets & Documents

- Related Issue #https://github.com/forem/forem-internal-eng/issues/490
- Closes #https://github.com/forem/forem-internal-eng/issues/490

## QA Instructions, Screenshots, Recordings

https://github.com/forem/forem/pull/18459

Click on the "Create Account" buttons and see them being tracked using `Ahoy::Event`.last

https://www.loom.com/share/1db7b629fec34947b044fef8306efdbf

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/10pk8d1EoR7HrO/giphy.gif)
